fetch.js

```diff
@@ -1,23 +1,29 @@
-(function(self) {
+(function(root, factory) {
+  /* jshint strict:false */
+  /* globals define, module */
+  if (typeof define === 'function' && define.amd) {
+    define([], function() { return factory({}, root) })
+  } else if (typeof module === 'object' && module.exports) {
+    factory(module.exports, root)
+  } else {
+    root.WHATWGFetch = factory({}, root)
+  }
+})(typeof self !== 'undefined' ? self : this, function(exports, global) {
   'use strict';
 
-  if (self.fetch) {
-    return
-  }
-
   var support = {
-    searchParams: 'URLSearchParams' in self,
-    iterable: 'Symbol' in self && 'iterator' in Symbol,
-    blob: 'FileReader' in self && 'Blob' in self && (function() {
+    searchParams: 'URLSearchParams' in global,
```

Comment:
To keep the diff to a minimum, could `global` simply continue to be called `self` like before?

Reply:
done